### PR TITLE
Improvement/cache structure

### PR DESCRIPTION
**Pull request by jgoerner**
# **TL;DR**
This PR optimizes the cache structure for `shortURL -> longURL` lookups

# About

Going through the code (mainly the `shortener.service.ts`) I've tried to see how the links are persisted. As far as I understood you're basically caching the pairs via [`cache-manager`](https://www.npmjs.com/package/cache-manager) in the following structure:

image

Within a cache in general, data can be looked up in two direction (at different costs). The cheap direction is from Key to Value as the cache is optimised to work on keys.

Looking up a key from value is possible, however (in the worst case) it requires to get **all the keys** and the value for **each key** - like it's implemented in the [`getOriginalUrl` method](https://github.com/origranot/url-shortener/blob/dd5979e840b851371aa8d886f7b34c6c38f35455/src/shortener/shortener.service.ts#L26-L32). In a way it is similar to a [Full Table Scan](https://en.wikipedia.org/wiki/Full_table_scan).

Without changing the underlying key-value data structure, one of those lookups will be cheap (`key -> value`) while the other will be costly (`value -> key`) and becoming more and more costly the more data is within the cache, as the full cache has to be scanned.

What I understood as the main use case for an URL shortener is to create a shortened URL (once), share the shortened link and lookup the shortened URL (multiple times).

That is the reason I'd suggest to flip key & value within the cache:

image

That structure is optimised for doing the lookup `shortURL -> longURL`.
 

# Other changes
One second @jgoerner, I've noticed you also removed the `getShortUrl` function. Don't we end in a situation in wich multiple short URLs could point to the same long URL?

**short answer:** Yes.
**sneaky answer:** Does it actually matter?

### CODE REVIEW

**Comment by origranot** [comment]
@jgoerner I waited so much for someone to notice that 💯 

I'm agree with you about the implementation, it's better when key is short URL - O(1).
It's not really matter if two short links are pointing to the same website, so I will approve it :)

Thank you so much!
